src/deep-cache/lib/SharedCache.js

```diff
@@ -70,11 +71,61 @@ export class SharedCache {
    * @returns {String}
    */
   buildKeyFromLambdaRuntime(runtime) {
-    let payload = JSON.stringify(runtime.request.data);
+    return runtime.context && runtime.context.has('invokedFunctionArn') ?
+      `${runtime.context.getOption('invokedFunctionArn')}#${SharedCache._payloadHash(runtime.request.data)}` :
+      null;
+  }
+
+  /**
+   * @param {Object} payload
+   * @returns {String}
+   * @private
+   */
+  static _payloadHash(payload) {
+    return SharedCache._hash(SharedCache._stringifyPayload(payload));
+  }
+
+  /**
+   * @param {Object} payload
+   * @private
+   */
+  static _stringifyPayload(payload) {
+    return JSON.stringify(
+      SharedCache._normalizeObject(payload)
+    );
+  }
+
+  /**
+   * @param {String} text
+   * @param {String} alg
+   * @returns {String}
+   * @private
+   */
+  static _hash(text, alg = 'sha1') {
+    return crypto.createHash(alg).update(text).digest('hex');
+  }
+
+  /**
+   * @param {Object} data
+   * @returns {Object}
+   * @private
+   */
+  static _normalizeObject(data) {
+    let normalizedData = {};
+
+    Object.keys(data).sort().forEach((key) => {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['let' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).](https://www.codacy.com/app/MitocGroup/deep-framework/file/1784383532/issues/source?bid=2895475&fileBranchId=3068904#l117)